api/app/signals/apps/api/v1/urls.py

```diff
@@ -93,5 +94,9 @@
 
         # Search
         re_path('search/?$', SearchView.as_view({'get': 'list'}), name='elastic-search'),
+
+        # csv
+        re_path('csv/?$', PrivateCsvViewSet.as_view({'get': 'download'}), name='download-csv'),
```

Comment:
Why not just register the `PrivateCsvViewSet`. In the `PrivateCsvViewSet` there is already a function declared as download. So this would be available. Or overwrite the `get` to do the download stuff.

Reply:
check